OpenMage/magento-lts
# GUI to Edit EAV Attributes & Sets - Base Code + Category

### Description (*)

This PR gives backend users the possibility to create/edit/delete any EAV attribute. In other words, all of these:

```sql
mysql> SELECT entity_type_id,entity_type_code FROM eav_entity_type ORDER BY entity_type_id;
+----------------+------------------+
| entity_type_id | entity_type_code |
+----------------+------------------+
|              1 | customer         |
|              2 | customer_address |
|              3 | catalog_category |
|              4 | catalog_product  |
|              5 | order            |
|              6 | invoice          |
|              7 | creditmemo       |
|              8 | shipment         |
+----------------+------------------+
```

I have only done light testing on this, we probably need more work before this can be merged as a useful feature.

### Related Pull Requests

This PR builds on top of the work done by @fballiano. It is meant to supersede these PRs if the community accepts.
1. #2260
2. #2264

### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Open the backend
2. See new options in Sales > Attributes, Catalog > Attributes, Customer > Attributes
3. Try to create new attributes or attribute sets

### Questions or comments

Todo:
- [x] Check translation strings
- **For another PR**: Modify customer attributes to add `use_in_forms` option?
- **For another PR**: Modify Customers > Manage Groups to allow us to select a attribute set for each customer group?
- **For another PR**: Block the user from editing/deleting "system" attributes. Use `adminhtml_[entityCode]_attribute_scopes` event to disallow changing scope perhaps?
- [x] Check ACL permissions, and make sure they didn't break BC with new Catalog > Attributes > Products > Attributes path.
- [x] Fit icon-head for `admin/XXX_attribute/new/` and `admin/XXX_set/add/`
- [ ] Figure out Copyright headers -- most of the code is copy/pasted from other core code, so I think we still need Magento, LLC copyright notice (in addition to ours in another PR).
- [x] Fix array short syntax
- [ ] Remove all prototypejs code
- [x] merge at least customer or category management PR into this, otherwise this is untestable
- **For another PR**: Review the SQL commands at the end of [this comment](https://github.com/OpenMage/magento-lts/pull/2264#issuecomment-1181890693). Should we add an upgrade script?


### Contribution checklist (*)
